spec/unit/views/index_as_blog_spec.rb

```diff
@@ -0,0 +1,76 @@
+require 'rails_helper'
+
+include ActiveAdmin
+describe ActiveAdmin::Views::IndexAsBlog do
+  subject { described_class.new }
+
+  describe '#build' do
+    let(:page_presenter) { double('page_presenter', block: nil) }
+    let(:collection) { double('collection') }
+
+    before do
+      expect(subject).to receive('build_posts')
+      expect(subject).to receive('add_class').with('index')
+    end
+
+    context 'when page_presenter has no block' do
+      before do
+        subject.build(page_presenter, collection)
+      end
+
+      it do
+        expect(subject.instance_variable_get(:@page_presenter))
+          .to eq(page_presenter)
+        expect(subject.instance_variable_get(:@collection)).to eq(collection)
+      end
+    end
+
+    context 'when page_presenter has block' do
+      let(:block) { Proc.new { double('proc_method') } }
+
+      before do
+        allow(page_presenter).to receive(:block).and_return(block)
+        allow(subject).to receive('instance_exec')
+        subject.build(page_presenter, collection)
+      end
+
+      it do
+        expect(subject).to have_received('instance_exec')
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.